src/network/transaction_request/mod.rs

```diff
@@ -18,88 +24,132 @@ pub struct TransactionRequest {
     pub fn set_factory_deps(&mut self, factory_deps: Vec<Bytes>) {
         self.eip_712_meta
             .get_or_insert_with(Eip712Meta::default)
             .factory_deps = factory_deps;
     }
 
+    /// Builder-pattern method for setting factory deps.
     pub fn with_factory_deps(mut self, factory_deps: Vec<Bytes>) -> Self {
         self.set_factory_deps(factory_deps);
         self
     }
 
+    /// Get the custom signature for the transaction.
+    pub fn custom_signature(&self) -> Option<&Bytes> {
+        self.eip_712_meta
+            .as_ref()
+            .and_then(|meta| meta.custom_signature.as_ref())
+    }
+
+    /// Set the custom signature  for the transaction.
     pub fn set_custom_signature(&mut self, custom_signature: Bytes) {
         self.eip_712_meta
             .get_or_insert_with(Eip712Meta::default)
             .custom_signature = Some(custom_signature);
     }
 
+    /// Builder-pattern method for setting custom signature.
     pub fn with_custom_signature(mut self, custom_signature: Bytes) -> Self {
         self.set_custom_signature(custom_signature);
         self
     }
 
-    pub fn zksync_deploy(
+    /// Get the paymaster params for the transaction.
+    pub fn paymaster_params(&self) -> Option<&PaymasterParams> {
+        self.eip_712_meta
+            .as_ref()
+            .and_then(|meta| meta.paymaster_params.as_ref())
+    }
+
+    /// Set the paymaster params for the transaction.
+    pub fn set_paymaster_params(&mut self, paymaster_params: PaymasterParams) {
+        self.eip_712_meta
+            .get_or_insert_with(Eip712Meta::default)
+            .paymaster_params = Some(paymaster_params);
+    }
+
+    /// Builder-pattern method for setting paymaster params.
+    pub fn with_paymaster_params(mut self, paymaster_params: PaymasterParams) -> Self {
+        self.set_paymaster_params(paymaster_params);
+        self
+    }
+
+    /// Builder-pattern method for building a ZKsync EIP-712 create2 tranasaction.
+    pub fn with_create2_params(
         self,
+        salt: B256,
         code: Vec<u8>,
         constructor_data: Vec<u8>,
         factory_deps: Vec<Vec<u8>>,
     ) -> Result<Self, BytecodeHashError> {
-        self.zksync_deploy_inner(None, code, constructor_data, factory_deps)
+        self.with_create_params_inner(Some(salt), code, constructor_data, factory_deps)
     }
 
-    pub fn zksync_deploy_with_salt(
+    /// Builder-pattern method for building a ZKsync EIP-712 create tranasaction.
+    pub fn with_create_params(
         self,
-        salt: B256,
         code: Vec<u8>,
         constructor_data: Vec<u8>,
         factory_deps: Vec<Vec<u8>>,
     ) -> Result<Self, BytecodeHashError> {
-        self.zksync_deploy_inner(Some(salt), code, constructor_data, factory_deps)
+        self.with_create_params_inner(None, code, constructor_data, factory_deps)
     }
 
-    fn zksync_deploy_inner(
-        mut self,
+    fn with_create_params_inner(
```

Comment:
I'd actually like to remove this inner method in lieu of inlining the code. create/create2 are different code paths and selecting them on the basis of salt is counter-intuitive.